tests/components/iotty/test_api.py

```diff
@@ -0,0 +1,79 @@
+"""Unit tests for iottycloud API."""
+
+from unittest.mock import patch
+
+from aiohttp import ClientSession
+import pytest
+
+from homeassistant.components.iotty import api
+from homeassistant.components.iotty.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_api_create_fail(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test API creation with no session."""
+
+    with pytest.raises(ValueError) as excinfo:
+        _ = api.IottyProxy(hass, None, None)
+    assert "websession" in str(excinfo.value)
+
+    with pytest.raises(ValueError) as excinfo:
+        _ = api.IottyProxy(hass, aioclient_mock, None)
+    assert "oauth_session" in str(excinfo.value)
+
+
+async def test_api_create_ok(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    aiohttp_client_session: None,
+    local_oauth_impl: ClientSession,
+) -> None:
+    """Test API creation."""
+
+    mock_config_entry.add_to_hass(hass)
+    assert mock_config_entry.data["auth_implementation"] is not None
+
+    config_entry_oauth2_flow.async_register_implementation(
+        hass, DOMAIN, local_oauth_impl
+    )
+
+    iotty = api.IottyProxy(hass, aiohttp_client_session, local_oauth_impl)
+
+    assert iotty is not None
```

Comment:
This is a meaningless assertion; a newly created Python object can't be `None`. If you just want to assert we can create the object, remove the assignment and assertion, and explain in the test's docstring that we're checking we can create an `IottyProxy` without raising.

Reply:
Totally agree! Done